### PR TITLE
fix: cl merge main swaprouter changes (state-breaking)

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

These changes are required to wire up Roman's PR of merging concentrated-liquidity-main branch to main. This PR is separated since it touches logic other than strictly the x/concentrated liquidity folder, and we should therefore scrutinize this PR more than the original merge to main PR.

## Brief Changelog

- Adds cl to app keepers
- Adds cl entry into swaprouter keeper
- Adds cl as an app module
- Adds cl to the v14 store upgrade handler (we may punt this to v15 but it exists there now due to e2e)
- Removes events emitted from the cl msg server
- Removes redundnat validation in x/swaprouter's `CreatePool`. Explanations for each added:
   * https://github.com/osmosis-labs/osmosis/pull/3876#discussion_r1058490778
   * https://github.com/osmosis-labs/osmosis/pull/3876#discussion_r1058496005

### CODE REVIEW

File: go.sum
Comment (p0mvn) [inline go.sum:40]: Why do we need these?

**Comment by p0mvn** [comment]
Please do not merge until: https://github.com/osmosis-labs/osmosis/pull/3868 is merged

**Review by p0mvn — APPROVED** [review]
LGTM once https://github.com/osmosis-labs/osmosis/pull/3868 is merged

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM! We should try to get this merged asap now that #3868 is merged so we don't diverge too much on other CL PRs to main

**Pull request merged**
